timeline-chart/src/layer/time-graph-axis-cursors.ts

```diff
@@ -15,8 +16,8 @@ export class TimeGraphAxisCursors extends TimeGraphLayer {
     }
 
     afterAddToContainer() {
```

Comment:
This comment applies to:
- TimeGraphAxis
- TimeGraphAxisCursors
- TimeGraphChartGrid
- TimeGraphChartSelectionRange
- TimeGraphNavigator

In these classes, afterAddToContainer() is called when the layer is added and initialized in ReactTimeGraphContainer's componentDidMount(). This adds the listeners for the current instance.

But every time render() is done (by any change of selection range of view range with the mouse), a new instance of these classes is created, many times as a new instance for every pixel mouse move during time selection. But for these instances, afterAddToContainer() is not called and no listeners are added.

So the listener handling happens only for the original instance. This instance gets destroyed when the time graph is closed and the listeners are removed.

Not sure there is an issue here, but I'm wondering what happens with all those instances created by render(). These instances never have their update() method called.

Reply:
I figured out why the constructor is called at each render. It's caused by how it is used in the https://github.com/theia-ide/theia-trace-extension where they are embedded in a react component and at each render a new instance is created. So, this needs to be fixed there.